taf/developer_tool.py

```diff
@@ -252,29 +256,35 @@ def generate_repositories_json(repo_path, targets_directory, namespace='',
     targets_relative_dir:
       Directory relative to which urls of the target repositories are set, if they do not have remote set
   """
+
   custom_data = _read_input_dict(custom_data)
   repositories = {}
-  auth_repo_targets_dir = os.path.join(repo_path, TARGETS_DIRECTORY_NAME)
-  for target_repo_dir in os.listdir(targets_directory):
-    repo_path = os.path.join(targets_directory, target_repo_dir)
-    if not os.path.isdir(repo_path):
+
+  repo_path = Path(repo_path).resolve()
+  auth_repo_targets_dir = repo_path / TARGETS_DIRECTORY_NAME
+  targets_directory = Path(targets_directory).resolve()
+  if namespace is None:
+    namespace = targets_directory.name
```

Comment:
After this line, `namespace` can't be `None`, so you can remove _if-checks_ in few lines below.

Reply:
Same as above, those _if_s are meant to check if `namespace` is an empty string